### PR TITLE
Adding support for storage of unauthenticated analytics

### PR DESCRIPTION
**Pull request by bhariharan**
Allows instantiation of an unauthenticated instance of `SalesforceAnalyticsManager` and storage of global events.

### CODE REVIEW

**Comment by khawkins** [comment]
@smcnulty-sfdc @chrlie

**Comment by brandonpage** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="true">Please re-submit this PR to the dev branch, we may have already fixed your issue.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">📱 Needs testing on a real device if change is non-trivial.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_PR-Check">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: libs/SalesforceSDK/src/com/salesforce/androidsdk/analytics/SalesforceAnalyticsManager.java [hunk 1]
```diff
@@ -95,23 +105,20 @@ public static synchronized SalesforceAnalyticsManager getInstance(UserAccount ac
...
-            return null;
-        }
-        String uniqueId = account.getUserId();
-        if (UserAccount.INTERNAL_COMMUNITY_ID.equals(communityId)) {
-            communityId = null;
-        }
-        if (!TextUtils.isEmpty(communityId)) {
-            uniqueId = uniqueId + communityId;
+        String uniqueId = UNAUTH_INSTANCE_KEY;
+        if (account != null) {
```
Comment (khawkins) [inline libs/SalesforceSDK/src/com/salesforce/androidsdk/analytics/SalesforceAnalyticsManager.java]: This is a subtle change to the previous behavior.  `getInstance(UserAccount, String)` would previously return the instance associated with the current user.  This changes the behavior to get the unauthenticated instance.  Do you foresee issues with this behavior change?

Comment (bhariharan) [reply libs/SalesforceSDK/src/com/salesforce/androidsdk/analytics/SalesforceAnalyticsManager.java]: Yeah, I thought about it. We probably shouldn't automatically be substituting `currentUser` anyway. I looked at the callers and everything passes in a user account. I will call it out explicitly in the doc too.

**Comment by brandonpage** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/forcedotcom/SalesforceMobileSDK-Android/blob/17831384769bb38b64dbdff96a42beed1152efa1/libs/SalesforceSDK/src/com/salesforce/androidsdk/analytics/SalesforceAnalyticsManager.java#L423">libs/SalesforceSDK/src/com/salesforce/androidsdk/analytics/SalesforceAnalyticsManager.java#L423</a> - Consider using <code>apply()</code> instead; <code>commit</code> writes its data to persistent storage immediately, whereas <code>apply</code> will handle it in the background</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/forcedotcom/SalesforceMobileSDK-Android/blob/17831384769bb38b64dbdff96a42beed1152efa1/libs/SalesforceSDK/src/com/salesforce/androidsdk/analytics/SalesforceAnalyticsManager.java#L447">libs/SalesforceSDK/src/com/salesforce/androidsdk/analytics/SalesforceAnalyticsManager.java#L447</a> - Consider using <code>apply()</code> instead; <code>commit</code> writes its data to persistent storage immediately, whereas <code>apply</code> will handle it in the background</td>
    </tr>
  </tbody>
</table>

# Tests results for SalesforceSDK

<p align="right" data-meta="generated_by_SalesforceSDK">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Pull request merged**